### DISCUSSION TITLE
Suggestion: Shop pack UI tidy-up and standardization

### DISCUSSION THREAD

**Discussion by NobleKangaroo** [gameserverapp/Platform#953] [Suggestions]
## Summary
#793 was a nice change, removing the need to click between tabs, but I would like to propose some suggestions to further enhance the shop pack UI, increasing its usability by a decent amount. Most of our shop packs are single-command packs, and I presume the same is likely the case for many others. As such, we could make some enhancements to better utilize some of the left column space, enhancing usability and user experience as the interface would be more "balanced" between columns and require less vertical scrolling.

Additionally, while we're making these changes, we should aim to standardize the look and feel of all cards on shop packs, ensuring they all have headers of a uniform color. Many headers have had a grey header added, which tends to contrast poorly against a white background. A proposed improvement to this is to simply remove the `.card--grey` class from cards, allowing card headers to default to their default blue. If the grey header background styling is strongly desired, an alternative would be to instead ensure that all cards uniformly use the `.card--grey` class, implementing grey card headers as the standard.

## Proposed changes
I'd like to propose the following UI changes on shop packs:
1. Move the "Presentation" card to the left column, above the "Actions" card
2. Add a header to the "Actions" card to standardize all cards as having a header
3. Remove grey card headers, defaulting to blue, so cards are uniform and stand out from each other

## Change preview
![shop-pack-ui-tidy-up](https://github.com/user-attachments/assets/8d784524-aa9f-461b-a98a-6ccd72ae0d8c)

## Extra info
The following works for a preview; however, it breaks the CodeMirror editor, likely due to event bindings, so this is only useful as a preview of proposed UI. You'll be able to make this permanent, but figured something like this might be useful for toying around in Chrome's dev tools, or for others to experiment with.
```js
// Get pre-existing elements
var $eleColLeft = $('#settings-form .col-md-8'),
    $eleCardActions = $eleColLeft.find('.card:nth-of-type(1)'),
    $eleColRight = $('#settings-form .col-md-4'),
    $eleCardSettings = $eleColRight.find('.card:nth-of-type(1)'),
    $eleCardSettingsFormGroups = $eleCardSettings.find('div.form-group'),
    $eleCardSettingsFormGroupsWhereSelect = $($eleCardSettingsFormGroups[$eleCardSettingsFormGroups.length - 2]).find('select[name="restriction_id"]'),
    $eleCardRestrictions = $eleColRight.find('.card:nth-of-type(2)'),
    $eleCardRestrictionsWhere = $eleCardRestrictions.find('select[name="restriction_id"]'),
    $eleCardRestrictionsProhibit = $($eleCardRestrictions.find('.form-group:last')),
    $eleCardPresentation = $eleColRight.find('.card:nth-of-type(3)'),
    $eleCardAdvanced = $eleColRight.find('.card:nth-of-type(4)'),
    $eleSaveBtn = $('h3.page-title > span.right > button');

// Change 1 - Move the "Presentation" card to the left column, before the "Actions" card
$eleCardPresentation.insertBefore($eleCardActions);

// Change 2 - Add a header to the "Actions" card to standardize all cards as having a header
$eleCardActions.prepend(
    $('<div>').addClass('card-header').append(
        $('<div>').addClass('caption').append(
            $('<h6>').text('Actions')
        )
    )
);

// Change 3 - Remove grey card headers, defaulting to blue, so cards are uniform and stand out from each other
$('.card--grey').removeClass('card--grey');
```

**Comment by mvaessen** [comment]
Thanks for the detailed info!

I'm not entirely sure about the suggested change though, as it kind of buries the action field. I do think that (the actions) is what people use most on those pages, and once the description is set, they most likely wont need that anymore. Hence why I put it further down.

Curious to see what other folks think. If it makes sense to change this, then we can do that for sure

**Reply by NobleKangaroo** [reply]
> Thanks for the detailed info!
> 
> I'm not entirely sure about the suggested change though, as it kind of buries the action field. I do think that (the actions) is what people use most on those pages, and once the description is set, they most likely wont need that anymore. Hence why I put it further down.
> 
> Curious to see what other folks think. If it makes sense to change this, then we can do that for sure

Sure, gotcha. Also, instead of the header being "Commands", perhaps "Actions" is better, as there are other actions than commands. I've updated my initial suggestion to reflect the verbiage.

As a second option, what about underneath? Mostly it would be to just pull "Presentation" from the right column to the left, but after "Actions" (instead of before) is a fair second option. I personally prefer the former, "Presentation" before "Actions" but either one is better than as-is today and can understand wanting to see actions over the pack's description and image.

Doing this would allow the "Presentation" card to sit further and further down as the shop pack matures, which would perhaps make sense with a more established, more complex shop pack as it'll be less likely to need much editing as the shop pack advances.

Here's a preview of what that would look like:

![shop-pack-ui-tidy-up-alt](https://github.com/user-attachments/assets/0951f25c-b15d-4e10-b643-860106e6d975)

**Comment by birdman3131** [comment]
I will add that the presentations being on the side is significantly more difficult to layout as compared to before as it is much narrower than it is on the actual website.

Before with it being its own tab it allowed line breaks. As it sits now it is somewhat difficult to tell what is line breaks and what is word wrapping. If the textbox was resizable this would not be as big of an issue but as it sits it is not great when editing longer descriptions.

**Reply by NobleKangaroo** [reply]
> I will add that the presentations being on the side is significantly more difficult to layout as compared to before as it is much narrower than it is on the actual website.
> 
> Before with it being its own tab it allowed line breaks. As it sits now it is somewhat difficult to tell what is line breaks and what is word wrapping. If the textbox was resizable this would not be as big of an issue but as it sits it is not great when editing longer descriptions.

That's a good point too. I also find myself, especially with packs with multiple paragraphs of text, unsure what's a line break and what's wrapping. This change would also help with that issue by bringing it to the col-8 side (8 of 12 columns wide) rather than the col-4 side (4 of 12 columns wide), giving it twice the editing width.
